File: 백준/Silver/1522. 문자열 교환/문자열 교환.py
import sys
input = sys.stdin.readline

string = list(input().strip())
Length = len(string)
cntA = string.count('a')
answer = 1e9

for i in range(0,Length):
  cntB = 0
  for j in range(i,i+cntA):
    if(string[j%Length] == 'b'):
      cntB+=1
  answer = min(answer,cntB)

print(answer)

File: 백준/Silver/1522. 문자열 교환/문자열 교환.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const target = input[0].split("")
const countOfA = target.filter((val) => val == "a").length

let answer = Number.MAX_SAFE_INTEGER
for (let idx = 0; idx < target.length; idx++) {
  let countOfB = 0
  for (let next = idx; next < idx + countOfA; next++) {
    if (target[next % target.length] == "b") {
      countOfB += 1
    }
  }
  answer = Math.min(answer, countOfB)
}

console.log(answer)